browser/ui/webui/brave_wallet/nft/nft_ui.cc

```diff
@@ -0,0 +1,54 @@
+/* Copyright (c) 2021 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#include "brave/browser/ui/webui/brave_wallet/nft/nft_ui.h"
+
+#include <string>
+
+#include "brave/common/webui_url_constants.h"
+#include "brave/components/nft_display/resources/grit/nft_display_generated_map.h"
+#include "components/grit/brave_components_resources.h"
+#include "content/public/browser/web_contents.h"
+#include "content/public/browser/web_ui_data_source.h"
+#include "ui/resources/grit/webui_generated_resources.h"
+
+namespace nft {
+
+UntrustedNftUI::UntrustedNftUI(content::WebUI* web_ui)
+    : ui::UntrustedWebUIController(web_ui) {
+  auto* untrusted_source = content::WebUIDataSource::Create(kUntrustedNftURL);
+  untrusted_source->SetDefaultResource(IDR_BRAVE_WALLET_NFT_DISPLAY_HTML);
+  untrusted_source->AddResourcePaths(
+      base::make_span(kNftDisplayGenerated, kNftDisplayGeneratedSize));
+  untrusted_source->AddFrameAncestor(GURL(kBraveUIWalletPageURL));
+  untrusted_source->AddFrameAncestor(GURL(kBraveUIWalletPanelURL));
+
+  // TODO(nvonpentz) Determine CSP. Commented below was copied from trezor_ui.cc
+  //
+  // untrusted_source->OverrideContentSecurityPolicy(
+  //     network::mojom::CSPDirectiveName::StyleSrc,
+  //     std::string("style-src 'unsafe-inline';"));
```

Comment:
We should resolve this TODO before merge.  Would someone from security please make a recommendation?

Reply:
Are we proxying all of the images here from the server or are some of them coming in via IPFS nodes directly using a different protocol other than HTTP? Asking to get a sense if we run into limitations for CSP depending on where the content is being hosted.

Reply:
I don't see any reason that we'd want to allow JS execution here from the style sheets. I'd prefere to see this set to either `"style-src 'none';"` or `"style-src 'self';"`

The downside I see to setting it to none is it may cause weird interactions with SVGs that have inline stylesheets, but I've not checked it yet.

Reply:
@kdenhartog  This is done